lib/_http_client.js

```diff
@@ -629,13 +633,15 @@ function responseKeepAlive(req) {
   socket.removeListener('error', socketErrorListener);
   socket.removeListener('data', socketOnData);
   socket.removeListener('end', socketOnEnd);
-  socket.once('error', freeSocketErrorListener);
-  // There are cases where _handle === null. Avoid those. Passing null to
-  // nextTick() will call getDefaultTriggerAsyncId() to retrieve the id.
-  const asyncId = socket._handle ? socket._handle.getAsyncId() : undefined;
-  // Mark this socket as available, AFTER user-added end
-  // handlers have a chance to run.
-  defaultTriggerAsyncIdScope(asyncId, process.nextTick, emitFreeNT, req);
```

Comment:
Why was this async_hooks part removed from here?

Reply:
@mcollina: so instead of calling `emitFreeNT` in a nextTick we call `responseKeepAliveNT` in the nextTick. Which is more correct, otherwise we remove the  `'error'` handler too early.